angular-route/templates/_.client.route.js

```diff
@@ -0,0 +1,5 @@
+$stateProvider.
+  state('<%= slugifiedName %>', {
+    url: '/<%= slugifiedRoutePath %>',
+    templateUrl: 'modules/<%= slugifiedModuleName %>/client/views/<%= slugifiedViewName %>.client.view.html'
+  }).
```

Comment:
Oh sorry, I thought you were actually referring to angular-route code..my mistake. Can you briefly detail all the possible scenarios and what happens? I just want to make sure we didn't miss one and there is nothing better for each scenario.